### PR TITLE
Add Child Home Visit Problem Solving Module

### LINKED ISSUE #26: Problem Solving
[Problem Solving](https://docs.google.com/spreadsheets/d/1dr8uhaF9KN09DzcQf6-l_VsC6jXCQ5ddj1cKnuymkto/edit#gid=179989320&range=345:345)

### PR DESCRIPTION
**Pull request by devcolonel**
Close #26

### CODE REVIEW

File: opensrp-chw/src/nacp/assets/json.form-sw/child_hv_problem_solving.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
+{
+  "count": "1",
+  "encounter_type": "Kutatua Changamoto",
```
Comment (maulidkategile) [inline opensrp-chw/src/nacp/assets/json.form-sw/child_hv_problem_solving.json]: There should be a consistency in naming the encounter type

File: opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java [hunk 2]
```diff
@@ -8,18 +8,26 @@
...
 
 import java.text.MessageFormat;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import timber.log.Timber;
 
 public class ChildHomeVisitInteractorFlv extends DefaultChildHomeVisitInteractorFlv {
+    private final HashMap<String, Boolean> problemSolingEvaluationResults = new HashMap<>();
```
Comment (maulidkategile) [inline opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java]: Cross-check the usage of this variable and its corresponding implementation

File: opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java [hunk 3]
```diff
@@ -267,4 +276,68 @@ public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
...
+            public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+                if (StringUtils.isBlank(child_playing_challenge)) {
+                    return BaseAncHomeVisitAction.Status.PENDING;
+                }
+
+                if (StringUtils.isNotBlank(child_playing_challenge)) {
+                    return BaseAncHomeVisitAction.Status.COMPLETED;
+                } else {
+                    return BaseAncHomeVisitAction.Status.PARTIALLY_COMPLETED;
+                }
```
Comment (maulidkategile) [inline opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java]: You can also show PARTIALLY_COMPLETED status when the parent is experiencing challenges in communicating with her child

File: opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java [hunk 4]
```diff
@@ -267,4 +276,68 @@ public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
                 .build();
         actionList.put(context.getString(R.string.anc_home_visit_observations_n_illnes), observation);
     }
+
+    private void evaluateProblemSolving() throws Exception {
+        problemSolingEvaluationResults.put(context.getString(R.string.child_problem_solving), false);
+        class ChildHVProblemSolvingHelper extends HomeVisitActionHelper {
```
Comment (maulidkategile) [inline opensrp-chw/src/nacp/java/org/smartregister/chw/interactor/ChildHomeVisitInteractorFlv.java]: It is recommended to use actionHelper package/folder for more readability

**Review by maulidkategile — DISMISSED** [review]
LGTM

**Review by issyzac — APPROVED** [review]

**Pull request merged**